File: 1-two-sum/two_sum.c
/**
 * Note: The returned array must be malloced, assume caller calls free().
 */
int* twoSum(int* nums, int numsSize, int target) {
    int i,j;
    int* result = malloc(sizeof(int)*2);
    result[0]=0;
    result[1]=0;
    for(i=0;i<numsSize;i++){
        for (j = i+1;j<numsSize;j++){
            if((nums[i]+nums[j]) == target){
                result[0]=i;
                result[1]=j;
                break;
            }
        }
    }
    return result;
    
}

File: 1-two-sum/two_sum.go
package leetcode1

func twoSum(nums []int, target int) []int {

	// 放map 里直接查找即可，这里不能有重复元素
	intmap := make(map[int]int)
	for i, n := range nums {
		intmap[n] = i
	}

	for i := 0; i < len(nums); i++ {

		if j, ok := intmap[target-nums[i]]; ok && j != i {
			return []int{i, j}
		}

	}

	return []int{}
}
